drivers/include/pms5003.h

```diff
@@ -0,0 +1,264 @@
+        uint16_t nc_pm_0p5;     /**< Number concentration of all particles <= 0.5µm [#/cm^3] */
+        uint16_t nc_pm_1;       /**< Number concentration of all particles <= 1µm [#/cm^3] */
+        uint16_t nc_pm_2p5;     /**< Number concentration of all particles <= 2.5µm [#/cm^3] */
+        uint16_t nc_pm_5;       /**< Number concentration of all particles <= 5µm [#/cm^3] */
+        uint16_t nc_pm_10;      /**< Number concentration of all particles <= 10µm [#/cm^3] */
+        uint16_t reserved;      /**< Reserved */
+    } values;
+    uint16_t buffer[13];        /**< Convenience uint16_t buffer */
+} pms5003_pm_t;
+
+/**
+ * @brief   Particulate matter sensor values
+ *
+ * Timestamp and set of measured particulate matter values as sent by the device
+ */
+typedef struct {
+    pms5003_pm_t pm;    /**< Particulate matter values */
+    uint32_t timestamp; /**< Timestamp for pm values */
+} pms5003_data_t;
+
+/**
+ * @brief     Particulate matter sensor UART frame parser buffer
+ *
+ *  Data is sent in bigendian so needs to be converted to host order.
+ */
+typedef struct {
+    uint8_t idx;                /**< Parsing index in bytes */
+    union __attribute__((packed)) {
+        be_uint16_t be[16];     /**< start + len + data (13) + crc */
+        uint8_t bytes[32];      /**< convenience uint8_t buffer */
+    } buf;
+} pms5003_parser_t;
+
+/**
+ * @brief   Available read modes
+ */
+typedef enum {
+    PMS5003_PASSIVE_MODE = 0,   /**< Passive reading mode (on demand) */
+    PMS5003_ACTIVE_MODE,        /**< Active reading mode (continuos) */
+} pms5003_mode_t;
+
+/**
+ * @brief   Definition of available sleep states
+ */
+typedef enum {
+    PMS5003_SLEEP = 0,          /**< Sleep */
+    PMS5003_RUNNING,            /**< Wake */
+} pms5003_sleep_t;
+
+/**
+ * @brief   PMS5003 device state
+ * @internal
+ */
+typedef enum {
+    PMS5003_STATE_IDLE = 0,                 /**< Currently parsing a frame */
+    PMS5003_STATE_READ_REQUEST,             /**< Currently parsing a frame */
+    PMS5003_STATE_MODE_ACTIVE_REQUEST,      /**< Parsed valid frame */
+    PMS5003_STATE_MODE_PASSIVE_REQUEST,     /**< Parsed valid frame */
+    PMS5003_STATE_SLEEP_REQUEST,            /**< Parsed invalid frame */
+    PMS5003_STATE_SLEEP,                    /**< Parsed invalid frame */
+    PMS5003_STATE_WAKEUP_REQUEST,           /**< Parsed invalid frame */
+} pms5003_state_t;
+
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    uart_t uart;        /**< The UART device */
+    uint32_t baudrate;  /**< The UART baudrate */
+    gpio_t reset_pin;   /**< Reset pin, active low */
+    gpio_t enable_pin;  /**< Enable pin, active high*/
+} pms5003_params_t;
+
+/**
+ * @brief   pms5003 callback type
+ */
+typedef void (*pms5003_callback_t)(void *);
+
+/**
+ * @brief   Device descriptor for the driver
+ */
+typedef struct {
+    pms5003_data_t data;                /**< Last valid data */
```

Comment:
I can get rid of this by passing it as a parameter to the callback as in @maribu branch.